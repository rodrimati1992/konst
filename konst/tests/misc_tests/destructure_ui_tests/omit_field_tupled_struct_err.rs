struct Tupled(String, String, String);

const fn to_bar(foo: Tupled) -> String {
    konst::destructure!{Tupled(bar) = foo}
    bar
}

const fn to_bar_2(foo: Tupled) -> String {
    konst::destructure!{Tupled(bar, ..) = foo}
    bar
}


fn main(){}