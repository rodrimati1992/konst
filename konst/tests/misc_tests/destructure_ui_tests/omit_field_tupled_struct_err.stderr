error: `..` patterns are not supported in top-level tuple struct patterns
 --> tests/misc_tests/destructure_ui_tests/omit_field_tupled_struct_err.rs:9:5
  |
9 |     konst::destructure!{Tupled(bar, ..) = foo}
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `$crate::__destructure__tuple_struct_field_names` which comes from the expansion of the macro `konst::destructure` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0425]: cannot find value `bar` in this scope
  --> tests/misc_tests/destructure_ui_tests/omit_field_tupled_struct_err.rs:10:5
   |
10 |     bar
   |     ^^^ not found in this scope

error: pattern requires `..` due to inaccessible fields
 --> tests/misc_tests/destructure_ui_tests/omit_field_tupled_struct_err.rs:4:5
  |
4 |     konst::destructure!{Tupled(bar) = foo}
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `konst::destructure` (in Nightly builds, run with -Z macro-backtrace for more info)
help: ignore the inaccessible and unused fields
 --> src/macros/destructuring.rs
  |
  |             (0, .. 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15)
  |               ++++
